packages/tc-schema-sdk/data-accessors/relationship-types.js

```diff
@@ -0,0 +1,40 @@
+const type = require('./type');
+
+module.exports = {
+	// todo move to object rest parameters for options when upgrading node
+	accessor(options = {}) {
+		const types = this.rawData.getTypes();
+
+		// Find root type names of rich relationships (newer designed)
+		const rootRichRelationshipTypeNames = types
```

Comment:
See comment elsewhere about keeping relationship types separate from types. Then this duck-typing won't be necessary